bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -689,6 +701,38 @@ TEST_F(BcStTest, dstValidateRealSourceListReported) {
   ASSERT_EQ(BCStateTran::FetchingState::NotFetching, stateTransfer_->getFetchingState());
 }
 
+// Validate a recurring source selection, during ongoing state transfer;
+TEST_F(BcStTest, validatePeriodicSourceReplacement) {
+  ASSERT_NO_FATAL_FAILURE(SendCheckpointSummaries());
+  mockedSrc_->ReplyAskForCheckpointSummariesMsg();
+  int batch_count = 0;
+  while (true) {
+    ASSERT_NO_FATAL_FAILURE(
+        AssertFetchBlocksMsgSent(testParams_.expectedFirstRequiredBlockNum, testParams_.expectedLastRequiredBlockNum));
+    // once the source is selected, adding sleep for more than source replacement time duration
+    if (batch_count < 2) {
```

Comment:
This doesn't make sense to me. Why do we need to add 5 seconds delay between request and reply? This does never happen in real world and shouldn't happen in happy flow test scenario.
you have the mock delays as part of the reply (2 millisec).
Adding sleeps here will trigger unwanted retransmissions..since no replay is received in 5000ms and I believe the retransmission timeout is smaller. This creates an unwanted scenario in test, where you involve replacement  due to retransmissions.
You first spoke on makeing sourceReplicaReplacementTimeoutMs=500ms when we had 1 on 1 chat, why is it now 5000ms? sounds too much to me.
We have approx 6-7 batches here, 1650 blocks needed to be fetched. with 2ms delay it will take 3 to 3.5 second to transfer all. 
If you reduce the replacement to 500 it should replace multiple times.
Also, the test should be general enough to work if I change number of blocks fetched.